File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for c in s:
            match c:
                case "(":
                    stack.append("(")
                case "{":
                    stack.append("{")
                case "[":
                    stack.append("[")
                case ")":
                    if len(stack) == 0 or stack[-1] != "(":
                        return False
                    stack.pop()
                case "}":
                    if len(stack) == 0 or stack[-1] != "{":
                        return False
                    stack.pop()
                case "]":
                    if len(stack) == 0 or stack[-1] != "[":
                        return False
                    stack.pop()
                
        return len(stack) == 0
        

File: Data Structures & Algorithms/daily-temperatures/submission-1.cs
public class Solution {
    public int[] DailyTemperatures(int[] temperatures) {
        
		int[] sol = new int[temperatures.Length];
		Stack<(int val, int index)> stack = new Stack<(int val, int index)>();

		for (int i = 0; i < temperatures.Length; i++) {
			if (stack.Count == 0) {
				stack.Push((temperatures[i], i));
				continue;
			}

			while (stack.Count > 0 && temperatures[i] > stack.Peek().val) {
				sol[stack.Peek().index] = i - stack.Peek().index;
				stack.Pop();
			}
			stack.Push((temperatures[i], i));
		}
		return sol;
    }
}
